packages/state-transition/src/util/blindedBlock.ts

```diff
@@ -92,7 +92,7 @@ export function signedBlindedBlockToFull(
 
   // state transition can't seem to handle executionPayloadHeader presense in merge block
   // so just delete the extra field we don't require
-  delete (signedBlock.message.body as {executionPayloadHeader?: ExecutionPayloadHeader}).executionPayloadHeader;
+  (signedBlock.message.body as {executionPayloadHeader?: ExecutionPayloadHeader}).executionPayloadHeader = undefined;
```

Comment:
Not a fan of the `performance/noDelete` rule, a property to `undefined` is not equivalent in some cases

e.g.
```node
>  Object.keys({})
[]
>  Object.keys({foo: undefined})
[ 'foo' ]
```
or when merging objects
```node
> {...{hello: "world"}, ...{}}
{ hello: 'world' }
> {...{hello: "world"}, ...{hello: undefined}}
{ hello: undefined }
```

sometimes you just wanna get rid of the property altogether

Reply:
I'm actually a fan of this one.  Not having delete makes turbofan optimization much better.  We should be paying attention to the usage.  I've noticed it when we go from one SSZ container to another and curious if there is a way to update the ssz to ignore `undefined` fields if they are not part of the container definition...

Reply:
> ssz to ignore undefined fields if they are not part of the container definitio

that's already the case, it only reads fields that are defined in the container, and ignores all others

> Not having delete makes turbofan optimization much better

need to carefully review all cases then that are in this PR, afterwards it should be fine, just don't like lint rules that change behavior

Reply:
Reverted the change for this rule from this PR. 

I am in favor of not using `delete`, but this should be done with more deliberate review in a separate PR.

Reply:
Let's do it in another PR if this has performance improvements we can take advantage of